api_tests/users/views/test_user_external_login.py

```diff
@@ -0,0 +1,59 @@
+import pytest
+from api.base.settings.defaults import API_BASE
+from osf.models import OSFUser
+from osf_tests.factories import UserFactory
+
+
+@pytest.mark.django_db
+class TestExternalLogin:
+
+    @pytest.fixture()
+    def user_one(self):
+        user = UserFactory()
+        user.set_password('password1')
+        user.auth = (user.username, 'password1')
+        user.save()
+        return user
+
+    @pytest.fixture()
+    def payload(self):
+        return {
+            'data': {
+                'attributes': {
+                    'email': 'freddie@mercury.com',
+                    'auth_user_external_first_login': True,
```

Comment:
@cslzchen I think we can't trust the user to send this info in a the JSON payload, should this information come from cas?

Reply:
Yeah, this looks suspicious though I need to double check. I am pretty confident that CAS is not involved here. I think some of the data comes from the session info instead of user inputs in the form (with the old flask V1 API).

Reply:
So in `external_login_email_post()`:

```python
form = ResendConfirmationForm(request.form)
...
external_id_provider = session.get('auth_user_external_id_provider', None)
external_id = session.get('auth_user_external_id', None)
fullname = session.get('auth_user_fullname', None) or form.name.data
service_url = session.get('service_url', None)
...
 if form.validate():
        clean_email = form.email.data
...
```

Only the name and email come from the form; the rest comes from session. Basically the way how this works is CAS tell OSF to set this special session (anonymous) during first time ORCiD SSO sign-in. Thus the V2 API should retrieve them from session instead of post request payload.